### ISSUE TITLE
Youtube Plugin with markers not displaying correctly

### ISSUE THREAD

**Issue by ChongoShaun** [spchuang/videojs-markers#69]
When loading video via the youtube plugin, only the last marker in the array is showing, and it is placed at the very beginning of the progress bar:

![screen shot 2017-09-07 at 11 11 50 am](https://user-images.githubusercontent.com/1319786/30173484-70a67f84-93bd-11e7-8bc4-bd9afb1e535d.png)

markers work when loading non-youtube video sources.

Any advice?

**Comment by ChongoShaun** [comment]
I HEARBY DECLARE MYSELF TO BE AN IDIOT!

My test source video was longer than the demo, and I was adjusting the width of the markers themselves. Turns out they were just not spaced out enough in the time calls. They were overlapping themselves and were only about 15 seconds apart. SORRY!

**Comment by ChongoShaun** [comment]
I can get it to work correctly if I turn autoplay on, but once its disabled, it looks just as it does in the screen shot.

**Comment by irugoy** [comment]
I'm facing same issue, apparently is an issue with the getDuration method in Youtube.js, as per the Youtube API it returns a number in seconds and videojs-markers expect duration in minutes.
I did a quick fix dividing the getDuration /60  and worked ok.

Youtube.js : line 560

**Before**
  duration: function() {
      return this.ytPlayer ? (this.ytPlayer.getDuration()/60) : 0;
    },

**After**
  duration: function() {
      return this.ytPlayer ? (this.ytPlayer.getDuration()/60) : 0;
    },

**Comment by vhmth** [comment]
@irugoy sweet! Thanks for the update here.  @ChongoShaun does that fix things for ya?

**Comment by irugoy** [comment]
In addition to the autoplay thing I've found another issue, when using the Youtube plugin it uses a getDuration method to return the length of the video. Apparently that info is not available for the videojs object if autoplay is not set and getDuration returns NaN, so markers could no be positioned correctly. The work around I did is to leave autoplay on and pause the video once start playing listening to the play event and at one.

video.one('play', function(){
   video.pause();
});
